### PR TITLE
Add FileExists and DeleteFile to Env

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

**Comment by UncP** [comment]
LGTM

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -2877,6 +2877,14 @@ void crocksdb_env_join_all_threads(crocksdb_env_t* env) {
   env->rep->WaitForJoin();
 }
 
+void crocksdb_env_file_exists(crocksdb_env_t* env, const char* path, char** errptr) {
+  SaveError(errptr, env->rep->FileExists(path));
+}
+
+void crocksdb_env_delete_file(crocksdb_env_t* env, const char* path, char** errptr) {
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/c.cc]: Why delete it via `Env`? Why not just use `fs::remove_file`?

Comment (zhangjinpeng1987) [reply librocksdb_sys/crocksdb/c.cc]: Because `file` can be any kind of types, for example memory sst file.

Comment (BusyJay) [reply librocksdb_sys/crocksdb/c.cc]: Can you provide a link to the explanation of memory SST file? Does it have a path?

Comment (huachaohuang) [reply librocksdb_sys/crocksdb/c.cc]: In RocksDB, `Env` is an interface to abstract some platform-dependent APIs.
For example, I can provide a `MemEnv`, which maintains a virtual file system in memory.
Then I can create a `SstFileWriter` with the `MemEnv` to generate SST file in memory.

Ref: https://github.com/facebook/rocksdb/blob/master/env/mock_env.h

Comment (BusyJay) [reply librocksdb_sys/crocksdb/c.cc]: Thanks for the information.

**Pull request merged**